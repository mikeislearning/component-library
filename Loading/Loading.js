import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, View } from 'react-native';

import { colors } from 'app/theme';

import styles from './Loading.styles';

export default class Loading extends React.Component {
  static defaultProps = {
    color: colors.red.medium,
    centered: true,
    horizontal: true,
    size: 'large',
  };

  static propTypes = {
    color: PropTypes.string,
    centered: PropTypes.bool,
    horizontal: PropTypes.bool,
    size: PropTypes.oneOf(['small', 'large']),
  };

  render() {
    const { centered, color, horizontal, size } = this.props;

    const containerStyle = [];
    if (horizontal) {
      containerStyle.push(styles.horizontal);
    }
    if (centered) {
      containerStyle.push(styles.centered);
    }

    return (
      <View style={containerStyle}>
        <ActivityIndicator size={size} color={color} />
      </View>
    );
  }
}
