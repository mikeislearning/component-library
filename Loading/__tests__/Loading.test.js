/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import Loading from '../Loading';

describe('<Loading />', () => {
  it('renders correctly', () => {
    const loading = render(<Loading />);
    const json = loading.toJSON();
    expect(loading).toMatchSnapshot();
    expect(json.type).toEqual('View');
    expect(json.children[0].type).toEqual('ActivityIndicator');
  });
});
