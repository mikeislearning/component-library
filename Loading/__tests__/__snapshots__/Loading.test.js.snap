// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#CC2E41"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
